### ISSUE TITLE
Keep images below text elements

### ISSUE THREAD

**Issue by xmatthewx** [mozilla/webmaker-core#716]
Current implementation allows users to tap an element to move it above all other elements. This works well, except when it doesn't. 

@flukeout proposes that we keep images in a lower group, text and links in a higher group. Users would still be able to tap and send an element to the front, but they'd stay within their respective groups. Images would never be in front of text or ~~images~~ buttons. 

This will create a small limitation on what users can create, while reducing heartache for the majority of users. A tap-and-hold context menu might offer more robust control later, but this seems like a good interim solution. 

cc @vazquez @gvn

**Comment by vazquez** [comment]
:+1: from me. It makes sense to categorize our elements into two hierarchical groups. It would solve a lot of headaches!

**Comment by xmatthewx** [comment]
A good issue for @alanmoo to look at if he clears his desk.

**Comment by xmatthewx** [comment]
bump back to 1.6?

**Comment by alanmoo** [comment]
Yeah. MozFest got in the way of this

**Comment by alanmoo** [comment]
It seems to me the main issue here is that elements are popping forward all the time, getting in the way of other elements. Images just happen to be the worst offenders because they're such big touch targets, is this correct? I'll examine solving it with groupings of elements, but I might also explore other possible solutions if that's ok; I feel like there may be better solutions that require less over-engineering.

**Comment by alanmoo** [comment]
To clarify, the following seems like an opportunity to fix:
- Tap a button/text element, and it is selected.
- Accidentally tap (and immediately release) an image right behind it, and that image comes to the front

It might be better if tapping the image behind it results only in "selecting" the image, and another tap brings it to the front, that way there are less false "bring to front" actions.

(Note that I'm not saying we shouldn't do what's stated in the original issue, I just want to properly solve the problem)

**Comment by xmatthewx** [comment]
Does a single tap on a unselected element move it to the front? It does. Don't think that was intended. I agree with your proposed change. Tap once to select. Tap a selected element to bring to front. That will reduce the number of accidents for sure. The accidents will still be mysterious and frustrating.

**Comment by alanmoo** [comment]
Yeah. Currently a single tap on any element brings it to the front, but a drag does not, so it's a bit...odd feeling.

**Comment by xmatthewx** [comment]
Top three most discussed UX topics in the past 9 months: zoom, swipe nav, z-index.  We talked about this a ton, and discussed many many solutions. 

I'm confident that the image layer is a solution worth testing. It's a good balance: simple mental model that enables a range project ideas while reducing frustrating accidents. Unless...

Let us know if engineering cost seems to outweigh benefit.

**Comment by alanmoo** [comment]
Please play with the changes from #850, I think it dramatically reduces frustrating accidents without the added logic that I feel might over-complicate things.

**Comment by gvn** [comment]
It feels weird to me to have different priorities for different element types. I can't think of any tool that does this and any time we add extra logic to UI it means users have to either intuit the functionality or we need extra FTU explaining it.

If #850 feels sufficient without adding the logic then it gets a big :+1:  from me.

**Comment by xmatthewx** [comment]
We can put this on hold for now. Wait and see.
